modules/gdscript/gdscript.cpp

```diff
@@ -1413,68 +1422,67 @@ void GDScript::_init_rpc_methods_properties() {
 	}
 }
 
-void GDScript::clear() {
+void GDScript::clear(GDScript::ClearData *p_clear_data) {
 	if (clearing) {
 		return;
 	}
 	clearing = true;
 
-	RBSet<GDScript *> must_clear_dependencies = get_must_clear_dependencies();
-	HashMap<GDScript *, ObjectID> must_clear_dependencies_objectids;
-
-	// Log the objectids before clearing, as a cascade of clear could
-	// remove instances that are still in the clear loop
-	for (GDScript *E : must_clear_dependencies) {
-		must_clear_dependencies_objectids.insert(E, E->get_instance_id());
+	GDScript::ClearData data;
```

Comment:
A comment in this section of code stating that this defines whether this is the "parent" which is responsible for destroying the scripts later would make sense, I think?

Reply:
In fact, maybe the name `initialized_clear_data` ought to be `is_responsible_for_clearing` or something that's a bit more obvious?

Reply:
I updated the code.